### ISSUE TITLE
feature_request(config): support YAML configuration format

### ISSUE THREAD

**Issue by Kristinita** [release-it/release-it#521]
### 1. Summary

It would be nice, if release-it will support other configuration formats like YAML or TOML.

### 2. Argumentation

1. JSON is format for bots, not for humans. Humans write `.release-it.json`, it not automatically generated file. It would be nice to have more convenient formats.
1. JSON doesn't support comments. Users need comments. Users may forget, what is the meaning of certain keys or values.
1. Unfortunately, you often change the format of `.release-it.json` file. For example, the file `release-it.json` has changed [**nearly everything since then, as I wrote an article about the release-it**](https://kristinita.github.io/IT-articles/Fastest-way-to-add-new-version-of-Sublime-Text-3-package#release-itjson) in April 2018. It would be nice to have comments [**in configuration file**](https://github.com/release-it/release-it/blob/HEAD/conf/release-it.json), to document changes.

### 3. Additional information

+ [**JSON as configuration files: please don't**](https://arp242.net/weblog/json_as_configuration_files-_please_dont)

### 4. Related issues

Examples:

+ [**Markdownlint**](https://github.com/DavidAnson/markdownlint/issues/118)
+ [**Stylus Supremacy**](https://github.com/ThisIsManta/stylus-supremacy/issues/31)

Thanks.

**Comment by webpro** [comment]
Thanks @Kristinita, I totally agree the JSON format isn't human-friendly. I was already toying with the idea of having a `conf/release-it.js` which is a regular script that exports the config object, allowing to document it better. Would be an easy first step. What do you think?

Regarding the (breaking) changes of this configuration file. The project has come a long way, and I have done many refactorings to get it to the point where it is now. Using deprecation warnings and leaving some time between major releases, I have tried my best to minimize the hassle for people to update the config (which is documented in the [CHANGELOG](https://github.com/release-it/release-it/blob/master/CHANGELOG.md)). I'm pretty confident the project is in a good state and configuration should be stable. And it would definitely have helped if the config contained comments in a simple JavaScript module (or e.g. YAML, although I'm not a big fan)

**Comment by Kristinita** [comment]
@webpro, what do you think about cosmiconfig integration? [**Citation**](https://github.com/davidtheclark/cosmiconfig#cosmiconfig) from official description:

> By default, Cosmiconfig will start where you tell it to start and search up the directory tree for the following:

> + a `package.json` property
> + a JSON or YAML, extensionless "rc file"
> + an "rc file" with the extensions `.json`, `.yaml`, `.yml`, or `.js`.
> + a `.config.js` CommonJS module

> For example, if your module's name is "soursocks", cosmiconfig will search up the directory tree for configuration in the following places:

> + a soursocks property in `package.json`
> + a `.soursocksrc` file in JSON or YAML format
> + a `.soursocksrc.json` file
> + a `.soursocksrc.yaml`, `.soursocksrc.yml`, or `.soursocksrc.js` file
> + a `soursocks.config.js` file exporting a JS object

1. cosmiconfig support YAML and `.js` configuration formats, which were discussed in our thread.
1. Users, that use release-it now and save configuration in `.release-it.json` or `package.json`, no need to migrate.

Thanks.

**Comment by webpro** [comment]
Eventually I came to the conclusion it's quite flexible yet easy for release-it to use cosmiconfig to load a config file. Mostly because people using different programming languages may not use/like JSON.

Released this in [v12.1.0](https://github.com/release-it/release-it/releases/tag/12.1.0). Also see https://github.com/release-it/release-it#configuration for docs.

Let me know whether this works for you. In any case, thanks for bringing this up, @Kristinita!
